Entwurf/Zuarbeit Entwurf.tex

```diff
@@ -0,0 +1,80 @@
+   \subsubsection{View}
+   Die View ist für die Darstellung die die Interaktion mit dem Nutzer zuständig, solbald der Nutzer zum Beispiel auf der Oberfläche einen Klick ausführt, wird dies sofort an den Controller weitergegeben, der wiederrum das Model kontaktiert.
+   \subsubsection{Controller}
+   Der Controller ist ein zentrales Element, dass die Daten der View entgegenimmt und dann ein Event behandelt, die vom Controller ausgelöst werden.
+   \subsection{Paketdiagramm des Grobentwurfs}
+   \begin{figure}
+    \includegraphics[scale=0.5]{Entwurf_PD_Grobentwurf.png}
+    \label{fig:Paketdiagramm_Grobentwurf}
+    \caption{Paketdiagramm Grobentwurf}
+   \end{figure} 
+
+   \subsection{Erläuterungen zum Paketdiagramm}
+   Wie man in Abbildung \ref{fig:Paketdiagramm_Grobentwurf} sehen kann, besitzt das Programm die Pakete Start,Student,Admin, Model und Repository.
+   Bis auf das Model und das Repostitory besitzt jedes Paket eine oder mehrere Views und Controller, die dann mit dem Model interagieren. Das Model kommuniziert dann mit dem Repository und leitet die Ergebnisse dann weiter an die Controller und Views.
+   \newpage
+\section{Feinentwurf}
+Der Feinentwurf wurde zur besseren Übersicht in mehrere Klassendiagramme unterteilt. Für jede Rolle, die im Programm möglich ist, befindet sich ein Klassendiagramm.
+Davon getrennt liegt das Repository. Es wird verwendet um Objekte auf Datenbankrelationen abzubilden und Datenbankrelationen auf Objekte abzubilden. Da wir uns aus Gründen der Wartbarkeit für das MVC-Konzept entschieden haben,hat jeder Controller eine View.
+Diese kommunizieren außerdem noch mit dem Model. Dieses besitzt als alleinige Komponente den Zugang zur Datenbank. Damit ist auch hier ohne großer Aufwand ein Austausch mit lokalen Daten möglich.
+Durch diese Gestaltung der Komponenten haben wir außerdem eine bessere Wiederverwendbarkeit von Komponenten geschaffen. 
+
+ \subsection{Die Klassendiagramme im Detail}
+ \subsubsection{Das Paket Start}
+ 
+ \begin{figure}
+  \includegraphics[scale=0.5]{Entwurf_KD_Feinentwurf_Start.png}
+  \label{fig:Klassendiagramm_Start}
+  \caption{Feinentwurf des Klassendiagramms Start}
+ \end{figure} 
+ 
+ Das Klassendiagramm in Abbildung \ref{fig:Klassendiagramm_Start} bildet den Auswahldialog des Programms ab. Hier kann ausgewählt werden, welche Rolle der Benutzer einnimmt. 
+ Daher gibt es hier nur eine View und einen Controller. Diese bilden die Rollenauswahl ab.
+ \subsubsection{Das Paket Student}
+ 
+ \begin{figure}
+\includegraphics[scale=0.25]{Entwurf_KD_Feinentwurf_Student.png}
+\label{fig:Klassendiagramm_Student}
+\caption{Feinentwurf des Klassendiagramms Student}
+ \end{figure} 
+
+ In dem Klassendiagramm in Abbildung \ref{fig:Klassendiagramm_Student} wird die Rolle des Studenten abgegbildet. Es ist in drei Controller und Views unterteilt. Eine zur Gruppenauswahl, für die Termine und eine für den Buchungsdialog.
```

Comment:
Das sollte dann (konsistent mit Start) in eine eigene `\subsubsection`

Reply:
Es wird nicht die Rolle des Studenten, sondern alle Views und Controllers der Studentensicht. Da sollte dann kurz erklärt werden, was welche Klasse macht, aber da sich die Klassen nochmal geändert haben, kannst du damit noch kurz warten

Reply:
Hab ich doch: `\subsubsection{Das Paket Student}`. Oder wie meinst du das ?

Reply:
Oh stimmt, habe ich nicht gesehen